### PR TITLE
The big refactor — part 4: Marie Kondo edition

### PR DESCRIPTION
**Pull request by rock3r**
This PR aims at making sense of the packages and modules in the codebase. The main changes are:
1. The main UI code is in the `org.jetbrains.jewel.ui` package, base/non-Jewel-dependent stuff is in org.jetbrains.jewel.foundation`
    * Components are in `.component` and their styling in `.component.styling`
    * State definitions are in `.foundation.state`
    * Annotations are in `foundation`
    * Everything else (apart from pre-existing foundation stuff) was moved to be a subpackage of `core`
2. Core has been renamed to `ui` (both module and package)
3. The `int-ui-core` module is no more, things are now either in `core` or `int-ui-standalone`
4. Foundation code has been split out into its own module
5. Packages have been realigned based on the core structure across all modules
6. A few minor cleanups here and there, getting rid of compiler warnings and inspections

Note: I ran the cleanup code action on the codebase, so hopefully there are no weird dangling imports etc, but I'll double check (and I hope the reviewers do too)

### CODE REVIEW

File: int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/styling/IntUiCheckboxStyling.kt [hunk 1]
```diff
@@ -10,57 +10,59 @@ import androidx.compose.ui.unit.dp
...
-import org.jetbrains.jewel.styling.CheckboxStyle
+import org.jetbrains.jewel.ui.painter.PainterProvider
 
 @Composable
-fun CheckboxStyle.Companion.light(
-    colors: CheckboxColors = CheckboxColors.light(),
-    metrics: CheckboxMetrics = CheckboxMetrics.defaults(),
-    icons: CheckboxIcons = CheckboxIcons.light(),
-) = CheckboxStyle(colors, metrics, icons)
+fun org.jetbrains.jewel.ui.component.styling.CheckboxStyle.Companion.light(
```
Comment (devkanro) [inline int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/styling/IntUiCheckboxStyling.kt:16]: Why using full name?

File: int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/styling/IntUiLinkStyling.kt [hunk 2]
```diff
@@ -8,43 +8,38 @@ import androidx.compose.ui.text.style.TextDecoration
...
+) = org.jetbrains.jewel.ui.component.styling.LinkStyle(colors, metrics, icons, textStyles)
 
 @Composable
-fun LinkStyle.Companion.dark(
-    colors: LinkColors = LinkColors.dark(),
-    metrics: LinkMetrics = LinkMetrics.defaults(),
-    icons: LinkIcons = LinkIcons.defaults(),
-    textStyles: LinkTextStyles = LinkTextStyles.dark(),
-) = LinkStyle(colors, metrics, icons, textStyles)
+fun org.jetbrains.jewel.ui.component.styling.LinkStyle.Companion.dark(
```
Comment (devkanro) [inline int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/styling/IntUiLinkStyling.kt:27]: Using full name here too

File: int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/styling/IntUiCheckboxStyling.kt (hunk 1 above)
Comment (rock3r) [reply int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/styling/IntUiCheckboxStyling.kt:16]: Slipped through!

**Review by devkanro — APPROVED** [review]
Awesome works

**Pull request merged**